samples/saga/batching/sample.md

```diff
@@ -0,0 +1,68 @@
+---
+title: Batch messages using Saga
+summary: Use batching when generating tasks using Saga
+reviewed: 2022-04-09
+component: Core
+related:
+- nservicebus/sagas
+---
+
+
+## Introduction
+
+This sample shows how to control the task (work) generation via batching of the work items through Sagas. This pattern of throttled message generation can be helpful when the the rate of generation needs to be controlled:
+
+- When lots of messages are generated as part of an incoming message, the amount of messages can further delay other messages in the same queue.
+- When the work as a whole is completed only when all messages are processed. If all messages are generated upfront and the work is considered 'cancelled' or 'failed', generated messages that are already in the queue would still have to be processed (e.g. no-op).
+
+Alternativly to avoid congesting the queue, messages can be generated and routed to a [specific endpoint](https://docs.particular.net/nservicebus/messaging/routing#make-instance-uniquely-addressable) (endpoint per work type), or [lower the concurrency rate](https://docs.particular.net/nservicebus/operations/tuning#configuring-concurrency-limit) to have the messages processed one by one.
```

Comment:
Concurrency limit is different from processing rate.
```suggestion
Alternatively to avoid congesting the queue, messages can be generated and routed to a [specific endpoint](https://docs.particular.net/nservicebus/messaging/routing#make-instance-uniquely-addressable) (endpoint per work type), or [lower the concurrency limit](https://docs.particular.net/nservicebus/operations/tuning#configuring-concurrency-limit) to have the messages processed sequentially.
```